NAudio/Wave/WaveInputs/WasapiCaptureRT.cs

```diff
@@ -148,6 +148,14 @@ private void InitializeCaptureDevice(IAudioClient audioClientInterface)
             if (waveFormat == null)
             {
                 waveFormat = audioClient.MixFormat;
+                if (mix.BitsPerSample == 32)
```

Comment:
Also, just checking bit depth is 32 is not always sufficient to know that you have IEEE float audio, since some soundcards can work with 32 bit PCM, so I also like to check the sample type as well.

Reply:
WaveFormatExtensible.cs
public WaveFormat ToStandaedWaveFormat
I have adapted the decision code of this class.